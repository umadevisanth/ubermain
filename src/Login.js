import { View, Text, SafeAreaView, Image, Icon, TouchableOpacity } from 'react-native';
import React from 'react';
import Globalstyle from './Globalstyle';  // Import global styles
import { FontAwesome } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';  // Import navigation hook
import { LinearGradient } from 'expo-linear-gradient';  // Import LinearGradient for background gradient

const Login = () => {
  // Login details object
  const login = {
  
    move: 'Move with saftey',
    start: 'Get Started'
  };

  const navigation = useNavigation();  // Get navigation object
  const jsonData = JSON.stringify(login);  // Convert login object to JSON string (though unused)

  // Function to navigate to PhoneNumber screen
  const handleNavigate = () => {
    navigation.navigate('PhoneNumber');  // Navigate to 'PhoneNumber' screen
  };

  return (
    // Using LinearGradient for background color
    <LinearGradient
      colors={['#1976D2', '#535AFF']}
      style={Globalstyle.main}>
      
      {/* Main container */}
      <View style={Globalstyle.loginviewtexts}>
        <View style={Globalstyle.wholecont}>
          <View style={Globalstyle.whole}>
            {/* Touchable opacity for back navigation */}
            <TouchableOpacity onPress={() => navigation.goBack('Account')}>
              <View style={Globalstyle.arrowMain}>
                {/* Logo image */}
                <Image
                  source={require('../assets/Group49.png')} 
                  style={Globalstyle.loginlogoimage}
                />
                {/* Main logo image */}
                <Image
                  source={require('../assets/Grouplogo.png')}
                  style={Globalstyle.loginimagestyle}
                />
              </View>
            </TouchableOpacity>
          </View>

          <View style={Globalstyle.iconwhole}>
            <TouchableOpacity>
              <View style={Globalstyle.icon}>
                {/* Safety text */}
                <Text style={Globalstyle.Login_saftey}>{login.move}</Text>
                {/* Shield icon image */}
                <Image
                  source={require('../assets/checkshield.png')}
                  style={Globalstyle.loginimageheight}
                />
              </View>
            </TouchableOpacity>
          </View>
        </View>

        <View style={Globalstyle.Login_btncont}>
          {/* Button to navigate to PhoneNumber screen */}
          <TouchableOpacity onPress={handleNavigate}>
            <View>
              {/* Get Started button image */}
              <Image
                style={Globalstyle.getstart}
                source={require('../assets/Group52.png')}
              />
            </View>
          </TouchableOpacity>
        </View>
      </View>
    </LinearGradient>
  );
}

export default Login;
