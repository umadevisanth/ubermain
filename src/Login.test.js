import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import Login from '../src/Login';

// Mock @expo/vector-icons
jest.mock('@expo/vector-icons', () => ({
  FontAwesome: 'FontAwesome', // Mock the specific icon component you're using
}));

// Mock useNavigation hook
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

describe('Login component', () => {
  it('renders correctly', () => {
    // Mock useNavigation behavior
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    // Render the Login component
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
