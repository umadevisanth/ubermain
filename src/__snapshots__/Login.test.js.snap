// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#276FDD",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "right": 40,
        "top": 230,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../assets/Groupimage.png",
        }
      }
      style={
        {
          "height": 70,
          "width": 85,
        }
      }
    />
    <View
      style={
        {
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 5,
          "height": 181,
          "width": 181,
        }
      }
    >
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 64,
            "paddingTop": 50,
            "textAlign": "center",
          }
        }
      >
        Uber
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        {
          "borderColor": "#fff",
          "borderRadius": 16.73,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "marginTop": 25,
          "top": 250,
          "width": 251,
        }
      }
    >
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 18,
            "left": 15,
            "padding": 10,
          }
        }
      >
        Move with saftey
      </Text>
      <FontAwesome
        aria-hidden="true"
        class="fa fa-car"
        name="car"
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 20,
            "left": 35,
            "top": 14,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "top": 450,
        "width": 340,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#EDF6FF",
            "fontSize": 25,
            "left": 60,
            "padding": 10,
          }
        }
      >
        Get Started
      </Text>
    </View>
    <FontAwesome
      aria-hidden="true"
      class="fa fa-arrow-right"
      name="arrow-right"
      style={
        {
          "color": "#EDF6FF",
          "fontSize": 20,
          "left": 100,
          "right": 25,
          "top": 20,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
